scala/data_structures/bloom_filter.scala

```diff
@@ -10,32 +10,34 @@ case class BloomFilter[A](
   private val bitset: BitSet,
   private val maxSize: Int,
   private val numberOfHashFunctions: Int,
-  private val hashSeed: Int
+  private val hashSeed: Int,
 ) {
 
-  private lazy val hashFunctions: Stream[String => Int] =
+  private lazy val hashFunctions: LazyList[String => Int] =
     (1 to numberOfHashFunctions)
-      .toStream
+      .to(LazyList)
       .map(i => (str: String) => abs(stringHash(str, hashSeed + i)) % maxSize)
 
-  def +=(item: A): BloomFilter[A] = {
+  def :+(item: A): BloomFilter[A] = {
     val itemString = item.toString
-    copy(bitset=(this.bitset ++ hashFunctions.map(_(itemString))))
+    copy(bitset=(bitset ++ hashFunctions.map(_(itemString))))
   }
 
-  def ++=(items: Seq[A]): BloomFilter[A] = items.foldLeft(this)(_ += _)
+  def ++(items: IterableOnce[A]): BloomFilter[A] = items.iterator.foldLeft(this)(_ :+ _)
```

Comment:
This allows to feed the bloom filter a much wider set of sequences, since IterableOnce is much more generic than Seq, and encompasses lazy sequences